src/utils.rs

```diff
@@ -8,57 +10,89 @@ use serenity::model::{
 use std::collections::HashMap;
 
 pub fn get_num_guilds(guilds: &Vec<GuildStatus>) -> String {
-    if guilds.len() == 1 {
-        String::from("1 guild")
-    } else {
-        format!("{:?} guilds", guilds.len())
-    }
+    format!(
+        "{} guild{}",
+        guilds.len(),
+        if guilds.len() > 1 { "" } else { "s" }
+    )
 }
 
 pub fn get_guild_human_members(members: &Vec<Member>) -> usize {
-    let human_members: Vec<Member> = members
-        .iter()
-        .filter(|member| !member.user.bot)
-        .map(|x| x.clone())
-        .collect();
-
-    human_members.len()
+    members.iter().filter(|member| !member.user.bot).count()
 }
 
 pub fn get_guild_online_members(
     members: &HashMap<UserId, Member>,
     presences: &HashMap<UserId, Presence>,
 ) -> usize {
-    let mut online_members: usize = 0;
-    for (user_id, presence) in presences.iter() {
-        if !members.get(user_id).expect("User not found").user.bot
-            && presence.status == OnlineStatus::Online
-        {
-            online_members += 1;
-        }
-    }
-
-    online_members
+    presences
+        .iter()
+        // Get all online user ids
+        .filter_map(|(user_id, presence)| match presence.status {
+            OnlineStatus::Online => Some(user_id),
+            _ => None,
+        })
+        // Remove members that are bots
+        .filter(|user_id| !members.get(user_id).expect("User not found").user.bot)
+        .count()
```

Comment:
I think you implementation is really good actually, but I generally find functional style code easier to read. [`filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) is a cool method generally